### PR TITLE
feat(chips): md-add-on-blur functionality

### LINKED ISSUE #3364: md-chips: Add a chip onInputBlur
I think it's a good experience for md-chips to convert outstanding text to a chip when input loses focus. 

Here is a use-case when it's relevant. User does not know that he needs to click enter to add a chip. When users clicks "Send email", no email has been captured.

### PR DESCRIPTION
**Pull request by DevVersion**
* Allows developers to convert the remaining input text into a new chip on input blur.
  This is useful for example, when having chips for email addresses.

Closes #3364.

### CODE REVIEW

**Comment by clshortfuse** [comment]
LGTM

**Comment by hschriek** [comment]
md-add-on-blur is only fired, when tabbing out of the chip input field, but not when leaving the field by clicking to another field

**Comment by DevVersion** [comment]
@hschriek Not true. We are just listening for a native `blur` event of the associated input element.

> 
image

FYI: My mouse normally doesn't look like that :smile:

**Comment by clshortfuse** [comment]
@DevVersion I think he means when you type dd, and click on the chip to the left, it doesn't blur the entire component, so the event doesn't get passed. I haven't tested the PR, but I think that's what he means.

**Comment by DevVersion** [comment]
@clshortfuse This isn't a problem. The chip will be added as expected.

**Comment by ThomasBurleson** [comment]
@EladBezalel - please review.

**Comment by topherfangio** [comment]
@DevVersion I think we need to enable this for the contact chips as well? Seems like that will be the most useful case.

Also (I can't easily test with the contact chips), does/should it automatically add if the input is invalid?

For instance, if you use `md-add-on-blur="true"` with email chips, and the user enters `something@somewhere` but leaves off the `.com`, should we still add it?

**Comment by DevVersion** [comment]
@topherfangio As we discussed on Slack, we are now checking the validity of the `chips` ngModel and also of the `userNgModel` (if provided).

That was a great suggestion. Thanks again.

File: src/components/chips/js/chipsDirective.js [hunk 1]
```diff
@@ -224,7 +226,6 @@
         readonly: '=readonly',
         removable: '=mdRemovable',
         placeholder: '@',
-        mdEnableChipEdit: '@',
```
Comment (topherfangio) [inline src/components/chips/js/chipsDirective.js]: Curious; why was this line removed? Looks like we're still using it above?

Comment (DevVersion) [inline src/components/chips/js/chipsDirective.js]: Good point - it's because the scope binding is never used and we are only checking the attribute once by using `$attrs`.
> https://github.com/DevVersion/material/blob/3f20904c387e4c7154d67a73d5a8f94213a3607e/src/components/chips/js/chipsController.js#L57

**Comment by ThomasBurleson** [comment]
@jelbourn - gatekeeper ping.
